src/BigIntMsg.chpl

```diff
@@ -62,18 +62,26 @@ module BigIntMsg {
                 var retList: list(string);
                 var block_size = 1:bigint;
                 block_size <<= 64;
-                while || reduce (tmp!=0) {
-                  var low: [tmp.domain] bigint;
-                  // create local copy, needed to work around bug fixed in Chapel, but
-                  // needed for backwards compatability for now
-                  forall (lowVal, tmpVal) in zip(low, tmp) with (var local_block_size = block_size) {
-                    lowVal = tmpVal % local_block_size;
-                  }
+                if && reduce (tmp == 0) {
+                  // early out if we are already all zeroes
                   var retname = st.nextName();
-
-                  st.addEntry(retname, new shared SymEntry(low:uint));
+                  st.addEntry(retname, new shared SymEntry(tmp:uint));
                   retList.append("created %s".format(st.attrib(retname)));
-                  tmp /= block_size;
+                }
+                else {
+                  while || reduce (tmp!=0) {
```

Comment:
Just a formatting thing that sometimes comes up, `tmp != 0`

Reply:
So this is actually a case where the parenthesis are needed. I've messed this up enough times to have it seared in my brain, but if you do
```chapel
|| reduce tmp !=0
```
chapel will interpret it as `(|| reduce tmp) != 0` not `|| reduce (tmp != 0)`

i've def been confused by this before, but it's just the design choice the made make `reduce/scan` higher precedent than binops. Brad pointed me to this chapel issue about it: https://github.com/chapel-lang/chapel/issues/11463